Tests/test_cards.gd

```diff
@@ -98,10 +98,16 @@ func test_draw_to_max():
 
 func test_discard_specific():
 	_card_container.draw_cards(5)
-	var card: CardWorld = _card_container.cards_in_hand[2]
-	assert_eq(card.card_data.card_title, "Card3")
 	
-	_card_container.discard_card(card)
+	var discard_card: CardWorld = _card_container.cards_in_hand[2]
+	
+	assert_eq(_card_container.cards_in_hand[0].card_data.card_title, "Card1")
+	assert_eq(_card_container.cards_in_hand[1].card_data.card_title, "Card2")
+	assert_eq(discard_card.card_data.card_title, "Card3")
+	assert_eq(_card_container.cards_in_hand[3].card_data.card_title, "Card4")
+	assert_eq(_card_container.cards_in_hand[4].card_data.card_title, "Card5")
+	
+	_card_container.discard_card(discard_card)
```

Comment:
this should be before the `assert_eq` and the lines after should be:
```
        assert_eq(_card_container.cards_in_hand[0].card_data.card_title, "Card1")
	assert_eq(_card_container.cards_in_hand[1].card_data.card_title, "Card2")
	assert_eq(_card_container.cards_in_hand[2].card_data.card_title, "Card4")
	assert_eq(_card_container.cards_in_hand[3].card_data.card_title, "Card5")
```
This test is to make sure cards are discarded properly, which is not what is done here if you test before discarding

Reply:
Oh I see what you mean now. I'll fix this

Reply:
Should be good now